### PR TITLE
Update BinarySearch.js

### PR DESCRIPTION
**Pull request by Askanders**
The old algorithm didn't work, I believe for two main reasons:
1 -  Number.MAX_VALUE is not a valid array index as it is used to represent the highest possible value in javascript (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_VALUE);

2 - splice() is not a pure function, every time it is called it has the side effect of modifying the original array (https://www.w3schools.com/jsref/jsref_splice.asp) ;

So I rewrote the algorithm, it now returns an index ( -1 if not found ) and it works both on numbers and on strings.

# Welcome to JavaScript community

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new JavaScript files are placed inside an existing directory.
* [x] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Search/BinarySearch.js [hunk 1]
```diff
@@ -1,26 +1,70 @@
...
-  var mid = Math.floor(arr.length / 2)
-  if (arr[mid] === i) {
-    console.log('match', arr[mid], i)
-    return arr[mid]
-  } else if (arr[mid] < i && arr.length > 1) {
-    binarySearch(arr.splice(mid, Number.MAX_VALUE), i)
-  } else if (arr[mid] > i && arr.length > 1) {
-    binarySearch(arr.splice(0, mid), i)
+function binarySearch(arr, x, low = 0, high = arr.length - 1) {
+  let mid = Math.floor(low + (high - low) / 2);
```
Comment (ruppysuppy) [inline Search/BinarySearch.js]:
```suggestion
  const mid = Math.floor(low + (high - low) / 2);
```

**Comment by ruppysuppy** [comment]
@Askanders, you can run ```$ standard MyFile.js``` as stated in [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md) to format the file instead of formatting it manually

**Comment by Askanders** [comment]
Yes, thank you @ruppysuppy. I have a lot of configuration to do so I haven't installed the package yet, but I'll definitely do it in the future.

**Comment by itsvinayak** [comment]
> Everything else looks fine to me, still I would appriciate if @itsvinayak would review it once

sure and thanks for helping @ruppysuppy

**Pull request merged**